src/core/renderController.ts

```diff
@@ -91,34 +91,23 @@ export module Core {
     export function flush() {
       if (_animationRequested) {
         // Layout
-        var toCompute = d3.values(_componentsNeedingComputeLayout);
-        toCompute.forEach((c) => c.computeLayout());
+        _componentsNeedingComputeLayout.values().forEach((c: Component) => c.computeLayout());
 
-        // Top level render.
-        // Containers will put their children in the toRender queue
-        var toRender = d3.values(_componentsNeedingRender);
-        toRender.forEach((c) => c._render());
+        // Top level render; Containers will put their children in the toRender queue
+        _componentsNeedingRender.values().forEach((c: Component) => c.render());
 
-        // now we are flushing
         _isCurrentlyFlushing = true;
-
-        // Finally, perform render of all components
-        var failed: {[key: string]: Component} = {};
-        Object.keys(_componentsNeedingRender).forEach((k) => {
+        var failed = new Utils.Set<Component>();
+        _componentsNeedingRender.values().forEach((c: Component) => {
           try {
-            _componentsNeedingRender[k]._doRender();
+            c._doRender();
           } catch (err) {
-            // using setTimeout instead of console.log, we get the familiar red
-            // stack trace
-            setTimeout(() => {
-              throw err;
-            }, 0);
-            failed[k] = _componentsNeedingRender[k];
+            // throw error with timeout to avoid interrupting further renders
+            window.setTimeout(() => { throw err; }, 0);
```

Comment:
Does this pass tslint? I thought we enforce newline after `{`. I prefer the old way.